services/QuillLMS/app/controllers/canvas_integration/lti_controller.rb

```diff
@@ -10,7 +10,7 @@ class LtiController < ApplicationController
 
     def launch
       @link_text = LAUNCH_TEXT
-      @link_url = canvas_integration_lti_sso_path(canvas_instance_url: params[:custom_canvas_api_baseurl])
+      @link_url = canvas_integration_lti_sso_path(canvas_instance_url: custom_canvas_api_baseurl, role: role)
```

Comment:
I have some concerns about this approach that are probably easier to chat through, i'll ping you.